lib/trino-arrow/src/main/java/io/trino/arrow/type/PicosecondTimeVector.java

```diff
@@ -0,0 +1,29 @@
+package io.trino.arrow.type;
+
+import org.apache.arrow.memory.BufferAllocator;
+import org.apache.arrow.memory.util.hash.ArrowBufHasher;
+import org.apache.arrow.vector.ExtensionTypeVector;
+import org.apache.arrow.vector.FieldVector;
+import org.apache.arrow.vector.complex.StructVector;
+
+public class PicosecondTimeVector extends ExtensionTypeVector<StructVector> {
```

Comment:
represent picosecond times with a microsecond time and an offset. arrow has support for arbitrary fixed size types, could instead use a 12 byte type. that might be less usable by clients that dont have support for these extension types, but would be more forward compatible if we can convince the arrow project to add picosecond times (same applies to picosecond timestamps and the pico precision zoned time and timestamps)